ivadomed/inference.py

```diff
@@ -61,6 +61,16 @@ def get_preds(context: dict, fname_model: str, model_params: dict, gpu_id: int,
     # Define device
     cuda_available, device = imed_utils.define_device(gpu_id)
 
+    # Assign '.onnx' or '.pt' model based on availability and device
+    # On GPU, '.pt' model is prioritized whereas '.onnx' model is prioritized on CPU.
+    if Path(fname_model + '.onnx').is_file():
+        if cuda_available and Path(fname_model + '.pt').is_file():
+            fname_model = fname_model + '.pt'
+        else:
+            fname_model = fname_model + '.onnx'
+    else:
+        fname_model = fname_model + '.pt'
```

Comment:
When I read this snippet of code, the first observation that came to mind was: 

> "If neither the `.pt` nor the `.onnx` model exists, then `fname_model` will still be assigned `{fname_model}.pt` because of the last `else:` block. That doesn't seem right!"

Then I kept reading the PR, and realized this couldn't happen because of:

https://github.com/ivadomed/ivadomed/blob/1a913e079e3472b5784a2140804a51233fa8e0a0/ivadomed/models.py#L1503-L1505

My takeaway here is that the logic is a little difficult to follow right now, because the file-parsing code is split into 2 separate places (`get_preds` and `get_model_filenames`). So, what if we combined the `.pt`/`.onnx`-selection logic with the existing logic inside the `get_model_filenames` function? 

(This also feels a little more intuitive to me -- my expectation is that `get_model_filenames` would return the model filename, as opposed to `get_preds` taking care of that logic.)

Reply:
That is a great point. I agree with you, it would be a lot clearer to choose onnx or pt directly in `get_model_filenames`, thanks!
The only downside I see is that we will need to import `imed_utils` in `models.py` to call the `define_device` function, but I think it's worth it.

Reply:
> The only downside I see is that we will need to import imed_utils in models.py to call the define_device function, but I think it's worth it.

Hmm... is that necessary, actually? 🤔 

It seems like it we only require the `cuda_available` property for the the `if`/`else` block. And, looking at [`define_device`](https://github.com/ivadomed/ivadomed/blob/master/ivadomed/utils.py#L117-L137), we this get directly from `torch`:

https://github.com/ivadomed/ivadomed/blob/860f44ff98eb58a8cce64a8bcc69b73df5a57b7f/ivadomed/utils.py#L128

Reply:
You are right, thank you! And torch is already imported in models. Perfect, I'm doing the changes now.